crates/database/src/solver_competition.rs

```diff
@@ -64,6 +64,24 @@ WHERE s.tx_hash = $1
     sqlx::query_as(QUERY).bind(tx_hash).fetch_optional(ex).await
 }
 
+pub async fn load_by_order_uid(
+    ex: &mut PgConnection,
+    order_uid: OrderUid,
+) -> Result<Option<LoadCompetition>, sqlx::Error> {
+    const QUERY: &str = r#"
+SELECT sc.json, sc.id, s.tx_hash
+FROM solver_competitions sc
+-- outer joins because the data might not have been indexed yet
+LEFT OUTER JOIN settlements s ON sc.id = s.auction_id
```

Comment:
A partially fillable order will be part of many competitions, so this doesn't work.

Reply:
Wouldn't sorting by auction_id desc help?